src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -1154,6 +1073,41 @@ private static OffsetDateTime getDateTimeFromLong( long epochTime ) {
         return OffsetDateTime.ofInstant( Instant.ofEpochMilli( epochTime ), ZoneId.systemDefault() );
     }
 
+    private WriteEvent clearOrDeleteEntities( UUID entitySetId, Set<UUID> entityKeyIds, DeleteType deleteType ) {
+        int numUpdates = 0;
+        long maxVersion = Long.MIN_VALUE;
+
+        // access checks for entity set and properties
+        final Map<UUID, PropertyType> authorizedPropertyTypes =
+                getAuthorizedPropertyTypesForDelete( entitySetId, Optional.empty(), deleteType );
+
+        Iterable<List<UUID>> entityKeyIdChunks = Iterables.partition( entityKeyIds, MAX_BATCH_SIZE );
+        for ( List<UUID> chunkList : entityKeyIdChunks ) {
+            Set<UUID> chunk = Sets.newHashSet( chunkList );
+
+            WriteEvent writeEvent;
+
+            if ( deleteType == DeleteType.Hard ) {
+                deleteAssociations( entitySetId, Optional.of( chunk ) );
+                writeEvent = dgm.deleteEntities(
+                        entitySetId,
+                        chunk,
+                        authorizedPropertyTypes );
+            } else {
+                clearAssociations( entitySetId, Optional.of( chunk ) );
+                writeEvent = dgm.clearEntities(
+                        entitySetId,
+                        chunk,
+                        authorizedPropertyTypes );
+            }
+
+            numUpdates += writeEvent.getNumUpdates();
+            maxVersion = Math.max( maxVersion, writeEvent.getVersion() );
```

Comment:
Version doesn't matter for hard deletes. For soft deletes using max version instead of min version will mean an auditor sees the state of the system after a bunch of operations instead of before.

The preferred way would seem to be to use one single version captured from before when the deletes started issuing. 

We should also consider that these operations are not a single transaction and should be audited as a compound operation or modified such that we issue all deletes in a single transaction.